From 2dc2d4894876ba05c97227d346e89a2d4adb9ac9 Mon Sep 17 00:00:00 2001
From: Karthikeyan K <karthikk@habana.ai>
Date: Fri, 20 May 2022 03:51:40 +0300
Subject: [PATCH] [SW-82717] rom: add rr config for unsecure systems

As in gaudi3, the complete LBW and HBW space will be blocked by
default by the HW, we would be unable to access any address
in the LBW/HBW space in recovery, even for unsecure systems(sec_en
efuse bit is 0).

During PO phase it is helpful to have a means to update firmware
using fw loader application's unsecured methodology. As this is
done when sec_en is 0, it does not pose security risk.

Change-Id: I0362aca68e39c23ea2202db381e570817913b5a9
---

diff --git a/arch/gaudi3/arc/security.c b/arch/gaudi3/arc/security.c
index f7f000a..e8d9c9e 100644
--- a/arch/gaudi3/arc/security.c
+++ b/arch/gaudi3/arc/security.c
@@ -941,12 +941,31 @@
 	regmap_writel(AXPROT_SEC_VAL, mmD0_PCIE_ELBI_RR_MSTR_IF_AXPROT_LBW_ARPROT_1_OVRD);
 }
 
+static void security_rr_recovery_unsecured_config(void)
+{
+	rr_ctx_lbw[RR_OPEN_ALL].rr_idx = HL_SEC_LBW_RR_SHORT_0;
+	security_rr_config_lbw_if_common(&rr_ctx_lbw[RR_OPEN_ALL]);
+
+	rr_ctx_lbw[RR_OPEN_ALL].rr_idx = HL_SEC_LBW_RR_SHORT_13;
+	security_rr_config_lbw_if_common(&rr_ctx_lbw[RR_OPEN_ALL]);
+
+	rr_ctx_hbw[RR_OPEN_ALL].rr_idx = HL_SEC_HBW_RR_SHORT_0;
+	security_rr_config_hbw_if_common(&rr_ctx_hbw[RR_OPEN_ALL]);
+
+	rr_ctx_hbw[RR_OPEN_ALL].rr_idx = HL_SEC_HBW_RR_SHORT_7;
+	security_rr_config_hbw_if_common(&rr_ctx_hbw[RR_OPEN_ALL]);
+}
+
 void security_rr_recovery_config(void)
 {
-	security_pcie_elbi_axprot_config();
-	security_prot_bits_config();
-	security_rr_recovery_lbw_config();
-	security_rr_recovery_hbw_config();
+	if (arcboot_rom_global.config_fuse & EFUSE_SYSCFG_SECURITY_ENABLE_MASK) {
+		security_pcie_elbi_axprot_config();
+		security_prot_bits_config();
+		security_rr_recovery_lbw_config();
+		security_rr_recovery_hbw_config();
+	} else {
+		security_rr_recovery_unsecured_config();
+	}
 }
 
 void check_and_enable_4b_termination(void) {}
